pulpcore/app/settings.py

```diff
@@ -379,6 +382,33 @@
     },
 )
 
+remote_user_backend_auth_validator = Validator(
+    "AUTHENTICATION_BACKENDS",
+    cont="django.contrib.auth.backends.RemoteUserBackend",
+)
+
+json_header_auth_class_restframework_validator = Validator(
+    "REST_FRAMEWORK__DEFAULT_AUTHENTICATION_CLASSES",
+    cont="pulpcore.app.authentication.JSONHeaderRemoteAuthentication",
+    when=remote_user_backend_auth_validator,
+)
+
+authentication_json_header_validator = Validator(
+    "AUTHENTICATION_JSON_HEADER",
+    startswith="HTTP_",
+    must_exist=True,
+    when=remote_user_backend_auth_validator & json_header_auth_class_restframework_validator,
+)
+
+authentication_json_header_jq_filter_validator = Validator(
+    "AUTHENTICATION_JSON_HEADER_JQ_FILTER",
+    must_exist=True,
+    when=remote_user_backend_auth_validator & json_header_auth_class_restframework_validator,
+)
```

Comment:
We cannot make it dependent on the one RemoteUserBackend. And we shouldn't have to. The rule reads like if you use the auth backend, it must be configured.

```suggestion

json_header_auth_class_restframework_validator = Validator(
    "REST_FRAMEWORK__DEFAULT_AUTHENTICATION_CLASSES",
    cont="pulpcore.app.authentication.JSONHeaderRemoteAuthentication",
    when=remote_user_backend_auth_validator,
)

authentication_json_header_validator = Validator(
    "AUTHENTICATION_JSON_HEADER",
    startswith="HTTP_",
    must_exist=True,
    when=json_header_auth_class_restframework_validator,
)

authentication_json_header_jq_filter_validator = Validator(
    "AUTHENTICATION_JSON_HEADER_JQ_FILTER",
    must_exist=True,
    when=json_header_auth_class_restframework_validator,
)
```